sdk/keyvault/TEST_RESOURCES_README.md

```diff
@@ -0,0 +1,130 @@
+# KeyVault Test Resources
+
+The KeyVault packages provide a common [Azure Resource Manager template][KVARM] to help developers set up the resources needed
+to run all of the integration tests in each of the KeyVault clients. This ARM template is used automatically by the CI pipelines
+that run during pull requests, just as much as it is used to help developers deploy their own resources on their personal
+Azure accounts for manual testing purposes.
+
+If you wish to use this ARM template to deploy resources on your own Azure personal or business account, you can click
+the following deploy button:
+
+[![](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-sdk-for-js%2Fmaster%2Fsdk%2Fkeyvault%2Ftest-resources.json)
+
+Keep in mind that deploying these resources constitutes a purchase of Azure services that will be billed to your active account.
+
+## What resources will be created
+
+The [ARM template][KVARM] used by the Deploy Button will create:
+
+- A single standard KeyVault, on the provided location, under the provided tenant, with a name based on the provided "Base Name".
+
+You can read more about KeyVault pricing details [here](https://azure.microsoft.com/en-us/pricing/details/key-vault/).
+
+## The form to deploy the KeyVault resources
+
+Once you click the "Deploy to Azure" button, you will be prompted to fill a form.
+In this guide we will go through the necessary steps to fill this form for the KeyVault clients for JavaScript.
+
+### Logging in
+
+Clicking the button will make Azure attempt to authenticate your Azure account. Make sure to log in using the Azure account that you want to be billed through.
+If you want to **create a new account**, you may follow this guide: [Create an Azure account][CreateAzureAccount].
+
+### The "Subscription" field
+
+The first field you will encounter is the "Subscription" field.
+It will have a drop down menu where you will be able to search for the specific Azure subscription
+you may want to use as the host of the resources you will create.
+
+**This field is required.**
```

Comment:
The ARM templates currently don't have a lot of information to help users follow through how to fill the properties. Although I agree we should make them better, discussing and following up on feedback on a Markdown file seems like a better initial approach to me.

Reply:
I agree with you in principle - the reason I think we should consider moving more of this into the template is because you'd have to flip back and forth between this readme and the actual portal. It's better to keep them in the one spot.

I just clicked through to get the full experience (love that button!) and TBH it doesn't seem all that confusing for the common properties.

Some things that I realize though:
* You should be able to [derive the tenant ID from their chosen subscription](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#subscription), right? So that's a parameter you could get rid of.
* Each parameter offers this:
  ```json
  "metadata": {
        "description": "The application client ID used to run tests."
   }
  ```
  And that "Description" field shows up in the tooltip directly next to the parameter in the deployment page. Perhaps we should enhance those a bit more so the user can just click the deploy button and go? 
  I know this is bad but I think quite a few people are just going to do that and not even read these documents.

Reply:
Interestingly enough, I can't add metadata for some fields, like subscription, resource group and location.

Reply:
I can't add links either.

Reply:
I have added more information in the properties that I was able to change from the ARM template! Let me know if this is better.